File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.js
function solution(my_string, queries) {
    let str = my_string;
    for (let [s, e] of queries) {
        str = str.slice(0, s) + str.slice(s, e + 1).split('').reverse().join('') + str.slice(e + 1);
    }
    return str;
}

File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.java
class Solution {
    public String solution(String my_string, int[][] queries) {
        StringBuilder sb = new StringBuilder(my_string);
        for (int[] q : queries) {
            int s = q[0], e = q[1];
            sb.replace(s, e + 1, new StringBuilder(sb.substring(s, e + 1)).reverse().toString());
        }
        return sb.toString();
    }
}
